drivers/adc/adc_ambiq.c

```diff
@@ -343,47 +384,56 @@ static int adc_ambiq_pm_action(const struct device *dev, enum pm_device_action a
 }
 #endif /* CONFIG_PM_DEVICE */
 
-#define ADC_AMBIQ_DRIVER_API(n)                                                            \
-	static const struct adc_driver_api adc_ambiq_driver_api_##n = {                        \
+#ifdef CONFIG_ADC_ASYNC
+#define ADC_AMBIQ_DRIVER_API(n)                                                                    \
+	static const struct adc_driver_api adc_ambiq_driver_api_##n = {                            \
+		.channel_setup = adc_ambiq_channel_setup,                                          \
+		.read = adc_ambiq_read,                                                            \
+		.read_async = adc_ambiq_read_async,                                                \
+		.ref_internal = DT_INST_PROP(n, internal_vref_mv),                                 \
+	};
+#else
+#define ADC_AMBIQ_DRIVER_API(n)                                                                    \
+	static const struct adc_driver_api adc_ambiq_driver_api_##n = {                            \
 		.channel_setup = adc_ambiq_channel_setup,                                          \
 		.read = adc_ambiq_read,                                                            \
 		.ref_internal = DT_INST_PROP(n, internal_vref_mv),                                 \
 	};
+#endif
 
-#define ADC_AMBIQ_INIT(n)                                                                  \
-	PINCTRL_DT_INST_DEFINE(n);                                                             \
-	ADC_AMBIQ_DRIVER_API(n);                                                               \
-	static int pwr_on_ambiq_adc_##n(void)                                                  \
-	{                                                                                      \
+#define ADC_AMBIQ_INIT(n)                                                                          \
+	PINCTRL_DT_INST_DEFINE(n);                                                                 \
+	ADC_AMBIQ_DRIVER_API(n);                                                                   \
```

Comment:
please do format updates in a separate commit

Reply:
I will fix tomorrow. Thank you!